src/programQueueProcessor/unit.test.ts

```diff
@@ -170,13 +217,121 @@ describe("kafka integration", () => {
           resolve();
         }, 30000);
       });
+      const totalEsDocuments_1 = (
+        await esClient.search({
+          index: ALIAS_NAME,
+          track_total_hits: true,
+        })
+      ).body?.hits?.total?.value;
+      expect(totalEsDocuments_1).to.equal(DB_COLLECTION_SIZE);
+
+      // 2. update TEST-CA by publishing a clinical event and a RDPC event:
+      await programQueueProcessor.enqueueEvent({
+        programId: TEST_CA,
+        type: programQueueProcessor.knownEventTypes.CLINICAL,
+      });
+      await programQueueProcessor.enqueueEvent({
+        programId: TEST_CA,
+        type: programQueueProcessor.knownEventTypes.RDPC,
+        rdpcGatewayUrls: [RDPC_URL],
+      });
+
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          resolve();
+        }, 30000);
+      });
+
+      // check if number of TEST-CA documents is expected:
+      const query_test_ca = esb
+        .requestBodySearch()
+        .query(esb.termQuery("programId", TEST_CA));
+
+      const test_ca_documents = (
+        await esClient.search({
+          index: ALIAS_NAME,
+          body: query_test_ca,
+        })
+      ).body?.hits?.total?.value;
+      expect(test_ca_documents).to.equal(testDonorIds.length);
+
+      // check if new rdpc data is relfected in TEST-CA
+      const hits = testDonorIds.map(async (donorId) => {
+        const esQuery = esb
+          .requestBodySearch()
+          .size(testDonorIds.length)
+          .query(esb.termQuery("donorId", "DO" + donorId));
+        const test_ca_hits: EsHit[] = await esClient
+          .search({
+            index: ALIAS_NAME,
+            body: esQuery,
+          })
+          .then((res) => res.body.hits.hits)
+          .catch((err) => {
+            return [];
+          });
+        return { donorId: donorId, hits: test_ca_hits } as {
+          donorId: string;
+          hits: EsHit[];
+        };
+      });
+
+      const totalHits = await Promise.all(hits);
+      console.log(
+        `expecting total hits of TEST-CA to be ${testDonorIds.length}`
+      );
+      expect(totalHits.length).to.equal(testDonorIds.length);
+
+      for (const test_ca_hit of await Promise.all(hits)) {
```

Comment:
seems more straightforward as:
```
const hits = await Promise.all(testDonorIds.map(async (donorId) => {...})
expect(hits.length).to.equal(testDonorIds.length);

for (const const test_ca_hit of hits){ ... }
```

Reply:
also `expect(hits.length).to.equal(testDonorIds.length);` will never fail because `.map` always returns an array of the same length as the original, or throw an error. We're catching the error with `.catch((err) => { return []; })`, so there's never a case where we don't get back a value... So the test doesn't really validate anything